src/pages/permissions/panels/GroupOrIdentityChangesTable.tsx

```diff
@@ -0,0 +1,187 @@
+import { Icon } from "@canonical/react-components";
+import { FC, useState } from "react";
+import { IdentitiesOrGroupsChangeSummary } from "util/permissionIdentities";
+import Tag from "components/Tag";
+import { LxdIdentity } from "types/permissions";
+import LoggedInUserNotification from "./LoggedInUserNotification";
+import { useSettings } from "context/useSettings";
+
+interface Props {
+  identityGroupsChangeSummary: IdentitiesOrGroupsChangeSummary;
+  groupIdentitiesChangeSummary: IdentitiesOrGroupsChangeSummary;
+  identities: LxdIdentity[];
+  initialGroupBy: "identity" | "group";
+}
+
+const generateRowsFromIdentityGroupChanges = (
+  identityGroupsChangeSummary: IdentitiesOrGroupsChangeSummary,
+  loggedInIdentityID: string,
+) => {
+  const identityIDs = Object.keys(identityGroupsChangeSummary);
+
+  const rows: JSX.Element[] = [];
+  for (const id of identityIDs) {
+    const groupChangesForIdentity = identityGroupsChangeSummary[id];
+    const identityLoggedIn = id === loggedInIdentityID;
+    let isFirstRow = true;
+    Array.from(groupChangesForIdentity.added).forEach((group, idx) => {
+      const rowKey = `${id}-${group}-added-${idx}`;
+      const row = (
+        <tr className={isFirstRow ? "first-row" : ""} key={rowKey}>
```

Comment:
It is probably better to only have one row per left hand side item, to avoid the behaviour below. Than we can also get rid of this wild first-row concept here.

![image](https://github.com/canonical/lxd-ui/assets/1155472/fab6ad3b-36ac-473e-8b47-326bb8d66d8d)

Reply:
nice, was able to simplify the code a lot here as well :+1: